goutubedl.go

```diff
@@ -639,6 +644,10 @@ func (result Result) DownloadWithOptions(
 		cmd.Args = append(cmd.Args, "--audio-format", options.AudioFormats)
 	}
 
+	if result.Options.Impersonate != "" {
+		cmd.Args = append(cmd.Args, "--impersonate", result.Options.Impersonate)
+	}
+
```

Comment:
Could you check if this one is needed? wonder if the impersonate stuff gets passed on vi the info json file

Reply:
Hmm, that's strange. It was not working without impersonate in result options.
Right now, yes, as I see it's not needed.

Reply:
Ok 👍 thanks for testing. Note that CI will probably fail because some tests using youtube sources which gets blocked. So we can probably merge anyway.